app/src/api/dataSource/ClientApiDataSource.ts

```diff
@@ -117,6 +146,18 @@ export class ClientApiDataSource implements ClientApi {
       },
       config,
     );
+    // @ts-expect-error: Property 'inner' does not exist on type 'RpcError'
+    if (response.error?.inner?.response?.data === 'Token not valid.') {
+      await getNewJwtToken({ refreshToken, getNodeUrl });
+      return {
+        error: {
+          message:
+            'Your session expired, but we have refreshed it. Please try again.',
+          code: 500,
+        },
+      };
+    }
```

Comment:
this is repeated throughout the file, separate it as a standalone function?

Reply:
Previous comment